.github/workflows/ci.yml

```diff
@@ -14,29 +14,27 @@ defaults:
     shell: bash
 
 jobs:
-  beefore:
-    name: Pre-test checks
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v3.1.2
+        with:
+          python-version: ${{ env.min_python_version }}
+      - uses: pre-commit/action@v3.0.0
+
+  docs:
```

Comment:
Is the docs target needed? On Briefcase, we've fallen back to using ReadTheDocs CI directly - would it be preferable to turn that on here and leave out the docs block entirely?

Reply:
Maybe (I don't actually know how the RTD CI works). But this PR is going to be big enough already, so I think it would be better to make that change in a separate one.

Reply:
I checked and it appears that RTD hasn't been building successfully for a while; so I figured we should probably fix that ASAP.